src/axom/sidre/tests/sidre_group.cpp

```diff
@@ -660,6 +660,152 @@ TEST(sidre_group, iterate_groups)
+
+    int nViews = 0;
+    for(auto& view : foo_group->views())
+    {
+      EXPECT_EQ(foo_group, view.getOwningGroup());
+      EXPECT_TRUE(endsWith(view.getName(), "_view"));
+      ++nViews;
+    }
+    EXPECT_EQ(numExpViews, nViews);
+  }
+
+  // const iterate through groups and views of 'foo' using range-for syntax
+  // starting from a non-const group
+  {
+    int nGroups = 0;
+    for(const auto& group : foo_group->groups())
+    {
+      EXPECT_EQ(foo_group, group.getParent());
+      EXPECT_TRUE(endsWith(group.getName(), "_group"));
+      ++nGroups;
+    }
+    EXPECT_EQ(numExpGroups, nGroups);
+
+    int nViews = 0;
+    for(const auto& view : foo_group->views())
+    {
+      EXPECT_EQ(foo_group, view.getOwningGroup());
+      EXPECT_TRUE(endsWith(view.getName(), "_view"));
+      ++nViews;
+    }
+    EXPECT_EQ(numExpViews, nViews);
+  }
+
+  // iterate through groups and views of 'foo' using range-for syntax
+  // starting from a const group
+  {
+    const auto* cfoo_group = ds.getRoot()->getGroup("foo");
+
+    int nGroups = 0;
+    for(const auto& group : cfoo_group->groups())
+    {
+      EXPECT_EQ(foo_group, group.getParent());
+      EXPECT_TRUE(endsWith(group.getName(), "_group"));
+      ++nGroups;
+    }
+    EXPECT_EQ(numExpGroups, nGroups);
+
+    int nViews = 0;
+    for(const auto& view : foo_group->views())
+    {
+      EXPECT_EQ(foo_group, view.getOwningGroup());
+      EXPECT_TRUE(endsWith(view.getName(), "_view"));
+      ++nViews;
+    }
+    EXPECT_EQ(numExpViews, nViews);
+  }
+
+  // iterate though groups and views of 'foo' using iterator syntax
+  {
+    int nGroups = 0;
+    for(auto it = foo_group->groups().begin(), itEnd = foo_group->groups().end();
+        it != itEnd;
+        ++it)
+    {
+      EXPECT_EQ(foo_group, it->getParent());
+      EXPECT_TRUE(endsWith(it->getName(), "_group"));
+      ++nGroups;
+    }
+    EXPECT_EQ(numExpGroups, nGroups);
+
+    int nViews = 0;
+    for(auto it = foo_group->views().begin(), itEnd = foo_group->views().end();
+        it != itEnd;
+        ++it)
+    {
+      EXPECT_EQ(foo_group, it->getOwningGroup());
+      EXPECT_TRUE(endsWith(it->getName(), "_view"));
+      ++nViews;
+    }
+    EXPECT_EQ(numExpViews, nViews);
+  }
+
+  // iterate though groups and views of 'foo' using const iterator syntax
+  {
+    int nGroups = 0;
+    for(auto it = foo_group->groups().cbegin(),
+             itEnd = foo_group->groups().cend();
+        it != itEnd;
+        ++it)
+    {
+      EXPECT_EQ(foo_group, it->getParent());
+      EXPECT_TRUE(endsWith(it->getName(), "_group"));
+      ++nGroups;
+    }
+    EXPECT_EQ(numExpGroups, nGroups);
+
+    int nViews = 0;
+    for(auto it = foo_group->views().cbegin(), itEnd = foo_group->views().cend();
+        it != itEnd;
+        ++it)
+    {
+      EXPECT_EQ(foo_group, it->getOwningGroup());
+      EXPECT_TRUE(endsWith(it->getName(), "_view"));
+      ++nViews;
+    }
+    EXPECT_EQ(numExpViews, nViews);
+  }
+}
+
```

Comment:
I think we need a test like this that uses the new iterator for the ListCollection.  I don't see any test coverage for that.

Reply:
Agreed -- actually, it doesn't look like we have any unit tests for the `ItemCollection` or its derived classes. I'll add some.

Reply:
Thanks, @kennyweiss , the `sidre_collections` test looks good.